components/brave_rewards/common/features.cc

```diff
@@ -41,4 +41,13 @@ BASE_FEATURE(kAnimatedBackgroundFeature,
              "BraveRewardsAnimatedBackground",
              base::FEATURE_DISABLED_BY_DEFAULT);
 
+BASE_FEATURE(kPlatformCreatorDetectionFeature,
+             "BraveRewardsPlatformCreatorDetection",
```

Comment:
If we preserve the old detection mechanism, let's try to find a way to disable it when the feature is enable.
I'm going to disable Greaselion if !rewards_enabled in a dedicated PR. Maybe we can just adjust the condition after.

Reply:
https://github.com/brave/brave-browser/issues/40872